src/compiler/parser.ts

```diff
@@ -5918,11 +5918,40 @@ namespace Parser {
                 expression = parseTokenNode<PrimaryExpression>();
             }
             else if (lookAhead(nextTokenIsDot)) {
-                // This is an 'import.*' metaproperty (i.e. 'import.meta')
+                // This is either specificallly an `import.meta.resolve(…)` call or more generally an 'import.*' metaproperty (i.e. 'import.meta')
                 nextToken(); // advance past the 'import'
                 nextToken(); // advance past the dot
-                expression = finishNode(factory.createMetaProperty(SyntaxKind.ImportKeyword, parseIdentifierName()), pos);
-                sourceFlags |= NodeFlags.PossiblyContainsImportMeta;
+                const identifierName = parseIdentifierName();
+
+                // TODO: use `parseRightSideOfDot`?
+                const foundImportMetaResolve = tryParse(() => {
+                    // `meta` and `resolve` are not JavaScript keywords, so we attempt to look for the direct string identifiers to recognizer `import.meta.resolve(…)`.
+                    if (identifierName.escapedText === "meta") {
+                        if (token() === SyntaxKind.DotToken) {
+                            nextToken(); // advance past the dot
+                            const nextIdentifierName = lookAhead(parseIdentifierName);
+                            if (nextIdentifierName.escapedText === "resolve") {
+                                if (lookAhead(() => nextToken() === SyntaxKind.OpenParenToken)) {
+                                    return true;
+                                }
+                            }
+                        }
+    
+                    }
+                    return false;
+                });
+
+                if (foundImportMetaResolve) {
+                    sourceFlags |= NodeFlags.PossiblyContainsDynamicImport;
+
+                    expression = {} as any; // TODO: calculate the expression. Here are some failed attempts:
+                    // expression = finishNode(factoryCreateCallExpression(factoryCreateExpressionStatement(), undefined, parseArgumentList()), pos) as any;
+                    // expression = parseTokenNode<PrimaryExpression>();
+                    // expression = finishNode(factoryCreateToken(SyntaxKind.ImportMetaResolveExpression), getNodePos()) as PrimaryExpression;

```

Comment:
I need help constructing `expression` to be an `ImportCall` matching the `ImportMetaResolveExpression` interface (i.e. with a `kind` type/value of `SyntaxKind.ImportMetaResolveExpression`).

I can't figure out how to do this based off of existing code, as the left-hand expression `import.meta.resolve` is not a token and the expression-related code in `types.ts` and `parser.ts` all see. Should this be an identifier? A `PropertyAccessExpression`? I'd appreciate any help in figuring out what to do.